assets/model_monitoring/components/src/action_analyzer/contracts/action.py

```diff
@@ -0,0 +1,143 @@
+
+        Args:
+            question(str): the input question of the flow.
+            answer(str): the output answer of the flow
+            debugging_info(str): the json string of debugging info in a span tree structure.
+            prompt_flow_input(str): the json str of prompt flow input.
+        """
+        self.question = question
+        self.answer = answer
+        self.debugging_info = debugging_info
+        self.prompt_flow_input = prompt_flow_input
+
+
+    def to_json_str(self) -> str:
+        """Convert an action sample object to json string."""
+        attribute_dict =  self.__dict__
+        json_out = {}
+        for key, val in attribute_dict.items():
+            json_out[convert_to_camel_case(key)] = val
+        return json.dumps(json_out)
+
+
+class Action():
+    """Action class."""
+
+    def __init__(self,
+                 action_type: ActionType, 
+                 description: str,
+                 confidence_score: float,
+                 query_intention: str,
+                 deployment_id: str,
+                 run_id: str,
+                 positive_samples: list[ActionSample],
+                 negative_samples: list[ActionSample]) -> None:
+        """Create an action.
+
+        Args:
+            action_type(ActionType): the action type.
+            description(str): the description of the action.
+            confidence_score(float): the confidence score of the action.
+            query_intention(str): the query intention of the action.
+            deployment_id(str): the azureml deployment id of the action.
+            run_id(int): the azureml run id which generates the action.
+            positive_samples(list[ActionSample]): list of positive samples of the action.
+            negative_samples(list[ActionSample]): list of negative samples of the action.
+        """
+        self.action_id = str(uuid.uuid4())
+        self.action_type = action_type
+        self.description = description
+        self.confidence_score = confidence_score
+        self.query_intention = query_intention
+        self.creation_time = str(datetime.datetime.now())
+        self.deployment_id = deployment_id
+        self.run_id = run_id
+        self.positive_samples = positive_samples
+        self.negative_samples = negative_samples
+
+
+    def to_json_str(self) -> str:
+        """Convert an action object to json str."""
+        attribute_dict =  self.__dict__
+        json_out = {}
+        for key, val in attribute_dict.items():
+            if key == "action_type":
+                json_out[convert_to_camel_case(key)] = val.name
+            # serialize the samples
+            elif key.endswith("_samples"):
+                json_val = str([v.to_json_str() for v in val])
+                json_out[convert_to_camel_case(key)] = json_val
+            else:
+                json_out[convert_to_camel_case(key)] = val
+        return json.dumps(json_out)
+
+
+class DebuggingInfo:
```

Comment:
where is this class used?

Reply:
is it the base class of IndexDebugInfo?

Reply:
sorry this one should be removed. I think the name of the IndexDebuggingInfo is a little bit misleading.  I just renamed it to RetrievalSpanData. So the 2 classes are totally different:
**DebuggingInfo** should be the span tree in the action samples.
**RetrievalSpanData** is just an internal class which contains retrieval span metadata. It is for extracting retrieval span info and explode from trace level to span level